src/mono/wasm/runtime/types.ts

```diff
@@ -173,6 +173,12 @@ export type CoverageProfilerOptions = {
     send_to?: string // should be in the format <CLASS>::<METHODNAME>, default: 'WebAssembly.Runtime::DumpCoverageProfileData' (DumpCoverageProfileData stores the data into INTERNAL.coverage_profile_data.)
 }
 
+/// Options to configure the event pipe session
+export interface EventPipeSessionOptions {
+    /// location on the VFS where the session trace will be saved
```

Comment:
VFS is good for us, but most developers would not know how to access it. Could we optionally lift the data from VFS and create download from it via [`<a href="data:application/octet-stream;` ](https://stackoverflow.com/a/18197341/129124)
On the API it could be `downloadTraceAfterSessionEnd:boolean` and work only in the browser. In node it could save directly to real disk.

it's just idea, not blocking the merge

Reply:
Yea I do something like that in the sample.  Easy enough to promote to an API

Reply:
I ended up going another way.  Added `getTraceBlob()` and `getTraceDataURI()` methods to the session object.

tracing and saving can be done like this:

```javascript
   const session = MONO.diagnostics.createEventPipeSession();
   session.start ();
   // do work
   session.stop ();
   const blob = session.getTraceBlob();
   const url = URL.createObjectURL (blob);
   downloadLink (url, "local-filename.nettrace");  // makes an <a download="..." href="object:..." /> and clicks on it
   URL.revokeObjectURL (url);
```

To be honest I'm not sure if the data URI one is needed.  This stuff depends on SharedArrayBuffer and web workers, so it needs newish browsers anyway.